Dockerfile

```diff
@@ -1,73 +1,139 @@
-	org.opencontainers.image.created="$CREATED"
-
-ENV CATALINA_HOME /usr/local/tomcat
-ENV PATH $CATALINA_HOME/bin:$PATH
-RUN mkdir -p "$CATALINA_HOME"
-WORKDIR $CATALINA_HOME
+FROM $DISTRIB_NAME${DISTRIB_MAJOR} AS TCNATIVE_BUILD
+ARG TCNATIVE_VERSION=1.2.31
+ARG TCNATIVE_SHA512=2aaa93f0acf3eb780d39faeda3ece3cf053d3b6e2918462f7183070e8ab32232e035e9062f7c07ceb621006d727d3596d9b4b948f4432b4f625327b72fdb0e49
+ARG APR_VERSION=1.7.0
+ARG APR_SHA256=48e9dbf45ae3fdc7b491259ffb6ccf7d63049ffacbc1c0977cced095e4c2d5a2
+ARG APR_UTIL_VERSION=1.6.1
+ARG APR_UTIL_SHA256=b65e40713da57d004123b6319828be7f1273fbc6490e145874ee1177e112c459
+ENV BUILD_DIR=/build
+ENV INSTALL_DIR=/usr/local
+ENV TCNATIVE_MIRRORS \
+        https://www.apache.org/dyn/closer.cgi?action=download&filename= \
+        https://archive.apache.org/dist \
+        https://www-us.apache.org/dist \
+        https://www.apache.org/dist
+SHELL ["/bin/bash","-c"]
+RUN mkdir -p {${INSTALL_DIR},${BUILD_DIR}}/{tcnative,libapr,apr-util}
+WORKDIR $BUILD_DIR
+RUN set -eux; \
+        for mirror in $TCNATIVE_MIRRORS; do \
+	        if curl -fsSL https://www.apache.org/dist/tomcat/tomcat-connectors/KEYS | gpg --import; then \
+			curl -fsSL https://www.apache.org/dist/apr/KEYS | gpg --import; \
+			active_mirror=$mirror; \
+			break; \
+		fi; \
+	done; \
+	[ -n "active_mirror" ]; \
+	\
+	for filetype in '.tar.gz' '.tar.gz.asc'; do \
+	   	curl -fsSLo tcnative${filetype} "${active_mirror}/tomcat/tomcat-connectors/native/${TCNATIVE_VERSION}/source/tomcat-native-${TCNATIVE_VERSION}-src${filetype}"; \
+	   	curl -fsSLo libapr${filetype} "${active_mirror}/apr/apr-${APR_VERSION}${filetype}"; \
+	   	curl -fsSLo apr-util${filetype} "${active_mirror}/apr/apr-util-${APR_UTIL_VERSION}${filetype}"; \
+	done; \
+	\
+	echo "$TCNATIVE_SHA512 *tcnative.tar.gz" | sha512sum -c -; \
+	echo "$APR_SHA256 *libapr.tar.gz" | sha256sum -c -; \
+	echo "$APR_UTIL_SHA256 *apr-util.tar.gz" | sha256sum -c -; \
+	\
+	gpg --batch --verify tcnative.tar.gz.asc tcnative.tar.gz && \
+        tar -zxf tcnative.tar.gz --strip-components=1 -C ${BUILD_DIR}/tcnative; \
+	# signing keys seem unavailable.... \
+	#if gpg --batch --verify libapr.tar.gz.asc libapr.tar.gz; then \
+	#	echo signature checked; \
+	#else \
+	#	keyID=$(gpg --batch --verify libapr.tar.gz.asc libapr.tar.gz 2>&1 | awk '/RSA\ /{print $NF}'); \
+	#	gpg --server hkp://keys.gnupg.net --recv-keys "$keyID"; \
+	#	gpg --batch --verify libapr.tar.gz.asc libapr.tar.gz; \
+	#fi && \
+        tar -zxf libapr.tar.gz --strip-components=1 -C ${BUILD_DIR}/libapr; \
+	# signing keys seem unavailable.... \
+	#if gpg --batch --verify apr-util.tar.gz.asc apr-util.tar.gz; then \
+	#	echo signature checked; \
+	#else \
+	#	keyID=$(gpg --batch --verify apr-util.tar.gz.asc apr-util.tar.gz 2>&1 | awk '/RSA\ /{print $NF}'); \
+	#	gpg --server hkp://keys.gnupg.net --recv-keys "$keyID"; \
+	#	gpg --batch --verify apr-util.tar.gz.asc apr-util.tar.gz; \
+	#fi && \
```

Comment:
is signatures check not working worth investigating?

otherwise I would remove all the commented lines